### ISSUE TITLE
Sort optional fields

### ISSUE THREAD

**Issue by Grawl** [azat-io/eslint-plugin-perfectionist#66]
### What rule do you want to change?

Props

### Describe the problem

I want to sort optional props to the bottom. 

### Code example

I mean that `required?: boolean` is supposed to be after `count: number` in Props type object. 

### Validations

- [X] Read the [docs](https://eslint-plugin-perfectionist.azat.io).
- [X] Check that there isn't [already an issue](https://github.com/azat-io/eslint-plugin-perfectionist/issues) that reports the same bug to avoid creating a duplicate.

**Comment by jasongitmail** [comment]
Same. Came here for this too. 

Using TypeScript and would like to have required parameters first and optional follow, then apply Perfectionists sorting rules inside of those groups.

**Comment by d8corp** [comment]
The same for default values

Actual:
```typescript jsx
const {
  a,
  b = c, // ??????
  c,
} = test
```

Expected:
```typescript jsx
const {
  a,
  c,
  b = c,
} = test
```

**Comment by caioedut** [comment]
+1

**Comment by azat-io** [comment]
Thank you for your issue!

This feature was implemented in 96cfcd8be3dee5870a14de1736e76af7632cc8b0 and released in [v3.0.0](https://github.com/azat-io/eslint-plugin-perfectionist/releases/tag/v3.0.0).

If you'd like, you can support the release with a retweet:
https://x.com/azat_io_en/status/1815367279191761054
